packages/Search/examples/bvh_driver/bvh_driver.cpp

```diff
@@ -259,26 +260,44 @@ int main( int argc, char *argv[] )
 {
     KokkosScopeGuard guard( argc, argv );
 
-    bool const throw_exceptions = false;
-    bool const recognise_all_options = false;
-    Teuchos::CommandLineProcessor clp( throw_exceptions,
-                                       recognise_all_options );
+    namespace bpo = boost::program_options;
+    bpo::options_description desc( "Allowed options" );
     int n_values = 50000;
     int n_queries = 20000;
     int n_neighbors = 10;
     int buffer_size = 0;
     std::string source_pt_cloud = "filled_box";
     std::string target_pt_cloud = "filled_box";
-    clp.setOption( "values", &n_values, "number of indexable values (source)" );
-    clp.setOption( "queries", &n_queries, "number of queries (target)" );
-    clp.setOption( "neighbors", &n_neighbors,
-                   "desired number of results per query" );
-    clp.setOption( "buffer", &buffer_size,
-                   "size for buffer optimization in radius search" );
-    clp.setOption( "source-point-cloud-type", &source_pt_cloud,
-                   "shape of the source point cloud" );
-    clp.setOption( "target-point-cloud-type", &target_pt_cloud,
-                   "shape of the target point cloud" );
+    // clang-format off
+    desc.add_options()
+        ( "help", "produce help message" )
+        ( "values", bpo::value<int>(&n_values)->default_value(50000), "number of indexable values (source)" )
+        ( "queries", bpo::value<int>(&n_queries)->default_value(20000), "number of queries (target)" )
+        ( "neighbors", bpo::value<int>(&n_neighbors)->default_value(10), "desired number of results per query" )
+        ( "buffer", bpo::value<int>(&buffer_size)->default_value(0), "size for buffer optimization in radius search" )
+        ( "source-point-cloud-type", bpo::value<std::string>(&source_pt_cloud)->default_value("filled_box"), "shape of the source point cloud"  )
+        ( "target-point-cloud-type", bpo::value<std::string>(&target_pt_cloud)->default_value("filled_box"), "shape of the target point cloud"  )
```

Comment:
I would align further, so that names under names, values under values, descriptions under descriptions. But that could make lines too long.